src/neo4j_genai/kg_construction/schema.py

```diff
@@ -0,0 +1,164 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+from typing import Any, Dict, List, Literal, Tuple
+
+from neo4j_genai.exceptions import SchemaValidationError
+from neo4j_genai.pipeline import Component, DataModel
+from pydantic import BaseModel, ValidationError, model_validator, validate_call
+
+
+class SchemaProperty(BaseModel):
+    name: str
+    # See https://neo4j.com/docs/cypher-manual/current/values-and-types/property-structural-constructed/#property-types
+    type: Literal[
+        "BOOLEAN",
+        "DATE",
+        "DURATION",
+        "FLOAT",
+        "INTEGER",
+        "LIST",
+        "LOCAL_DATETIME",
+        "LOCAL_TIME",
+        "POINT",
+        "STRING",
+        "ZONED_DATETIME",
+        "ZONED_TIME",
+    ]
+    description: str = ""
+
+
+class SchemaEntity(BaseModel):
+    """
+    Represents a possible node in the graph.
+    """
+
+    label: str
+    description: str = ""
+    properties: List[SchemaProperty] = []
+
+
+class SchemaRelation(BaseModel):
+    """
+    Represents a possible relationship between nodes in the graph.
+    """
+
+    label: str
+    description: str = ""
+    properties: List[SchemaProperty] = []
+
+
+class SchemaConfig(DataModel):
+    """
+    Represents possible relationships between entities and relations in the graph.
+    """
+
+    entities: Dict[str, Dict[str, Any]]
+    relations: Dict[str, Dict[str, Any]]
+    potential_schema: List[Tuple[str, str, str]]
```

Comment:
I'm wondering whether this should be renamed to potential triplets instead? Or is potential_schema fine?